File: src/main/kotlin/com/oktfolio/kite/security/authentication/PhoneAuthenticationToken.java
package com.oktfolio.kite.security.authentication;

import org.springframework.security.core.GrantedAuthority;

import java.util.Collection;

/**
 * @author oktfolio oktfolio@gmail.com
 * @date 2020/02/27
 */
public class PhoneAuthenticationToken extends AbstractKiteAuthenticationToken {

    public PhoneAuthenticationToken(Object principal, Object credentials) {
        super(principal, credentials);
    }

    public PhoneAuthenticationToken(Object principal, Object credentials, Collection<? extends GrantedAuthority> authorities) {
        super(principal, credentials, authorities);
    }

}

File: src/main/kotlin/com/oktfolio/kite/security/authentication/PhoneAuthenticationToken.kt
package com.oktfolio.kite.security.authentication

import org.springframework.security.core.GrantedAuthority


/**
 * @author oktfolio oktfolio@gmail.com
 * @date 2020/05/04
 */
class PhoneAuthenticationToken : AbstractKiteAuthenticationToken {
    constructor(principal: Any, credentials: Any?) : super(principal, credentials)
    constructor(principal: Any, credentials: Any?, authorities: Collection<GrantedAuthority?>?)
            : super(principal, credentials, authorities)
}